src/Catalyst.Core.Modules.Keystore.Tests/IntegrationTests/KeyApiTest.cs

```diff
@@ -0,0 +1,354 @@
+using Catalyst.Core.Lib.Config;
+using Catalyst.TestUtils;
+using Lib.P2P;
+using Makaretu.Dns;
+using NSubstitute;
+using Org.BouncyCastle.Crypto.Parameters;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Keystore.Tests.IntegrationTests
+{
+    public sealed class KeyApiTest : FileSystemBasedTest
+    {
+        private readonly IKeyStoreService _keyStoreService;
+
+        public KeyApiTest(ITestOutputHelper output) : base(output)
+        {
+            var dfsOptions = new DfsOptions(new BlockOptions(), new DiscoveryOptions(), new RepositoryOptions(FileSystem, Constants.DfsDataSubDir), Substitute.For<KeyChainOptions>(), Substitute.For<SwarmOptions>(), Substitute.For<IDnsClient>());
+            _keyStoreService = new KeyStoreService(dfsOptions);
+            _keyStoreService.SetPassphraseAsync(new SecureString()).Wait();
+        }
+
+        [Fact]
+        public async Task Self_Key_Exists()
+        {
+            const string name = "self";
+            var _ = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+
+            var keys = await _keyStoreService.ListAsync();
+            var self = keys.Single(k => k.Name == "self");
+
+            var me = await _keyStoreService.FindKeyByNameAsync("self");
+            Assert.Equal("self", self.Name);
+            Assert.Equal(me.Id, self.Id);
+        }
+
+        [Fact]
+        public async Task Export_Import()
+        {
+            const string name = "self";
+            var _ = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+            var password = "password".ToCharArray();
+            var pem = await _keyStoreService.ExportAsync(name, password);
+            Assert.StartsWith("-----BEGIN ENCRYPTED PRIVATE KEY-----", pem);
+
+            var keys = await _keyStoreService.ListAsync();
+            var self = keys.Single(k => k.Name == name);
+
+            await _keyStoreService.RemoveAsync("clone");
+            var clone = await _keyStoreService.ImportAsync("clone", pem, password);
+            Assert.Equal("clone", clone.Name);
+            Assert.Equal(self.Id, clone.Id);
+        }
+
+        [Fact]
+        public void Export_Unknown_Key()
+        {
+            var password = "password".ToCharArray();
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = _keyStoreService.ExportAsync("unknown", password).Result;
+            });
+        }
+
+        [Fact]
+        public async Task Import_Wrong_Password()
+        {
+            const string name = "self";
+            var _ = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+
+            var password = "password".ToCharArray();
+            var pem = await _keyStoreService.ExportAsync("self", password);
+
+            var wrong = "wrong password".ToCharArray();
+            ExceptionAssert.Throws<UnauthorizedAccessException>(() =>
+            {
+                var _ = _keyStoreService.ImportAsync("clone", pem, wrong).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)